src/ast.ts

```diff
@@ -271,12 +271,10 @@ export const rewriteResultsAtPath = (
 
   if (path.length === 1) {
     if (Array.isArray(curResults)) {
-      newResults[curPathElm] = curResults.map((_, index) => {
-        const newValue = callback(curResults, index);
-        return newValue;
-      });
-
-      return newResults;
+      return curResults.reduce(
+        (reducedResults, _, index) => callback(reducedResults, curPathElm, index),
```

Comment:
right, I see what you're doing here. We don't have enough info to rewrite child array just on the parent object and the key. I think this is the best fix we can do while keeping the style as close to the current style as possible.